### ISSUE TITLE
An internal error occurred while type checking file / Error performing analysis: RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by georgeivanyan** [microsoft/pylance-release#2373]
## Environment data

-   Language Server version: v2022.2.1
-   OS and version: Edition	Windows 10 Enterprise, Version	21H2, OS build	19044.1526
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.9 (venv)

## Expected behaviour
Can typecheck my file (unable to show it here, not open source) when using wrap method from the class

## Actual behaviour

Can't typecheck file. Pylance gets an error:  (1140) An internal error occurred while type checking file / (1140) Error performing analysis: RangeError: Maximum call stack size exceeded

## Logs

20-Python Language Server.log

```
[Error - 9:19:20 AM] (1140) An internal error occurred while type checking file "c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\src\data_application\fix_application.py": RangeError: Maximum call stack size exceeded
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:934:18)
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1019:29)
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1019:29)
    at Object.isFlowNodeReachable (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1066:20)
    at isFlowPathBetweenNodes (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:2480:31)
    at filter (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17314:30)
    at Array.filter (<anonymous>)
    at lookUpSymbolRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17304:77)
    at getTypeFromName (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:3654:33)
    at xe (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:863:30)
[Error - 9:19:20 AM] (1140) Error performing analysis: RangeError: Maximum call stack size exceeded
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:934:18)
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1019:29)
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1019:29)
    at Object.isFlowNodeReachable (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1066:20)
    at isFlowPathBetweenNodes (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:2480:31)
    at filter (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17314:30)
    at Array.filter (<anonymous>)
    at lookUpSymbolRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17304:77)
    at getTypeFromName (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:3654:33)
    at xe (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:863:30)
```

exthost.log:

```
[2022-02-11 09:12:02.663] [exthost] [error] Error: Request textDocument/hover failed with message: Maximum call stack size exceeded
	at c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.python-2022.0.1814523869\out\client\extension.js:67:139263
	at c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.python-2022.0.1814523869\out\client\extension.js:67:139557
	at Immediate.<anonymous> (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.python-2022.0.1814523869\out\client\extension.js:67:139920)
	at processImmediate (internal/timers.js:461:21)
```

## Code Snippet / Additional information

I started to experience this when I use decorator and wrap method is in a class.
For example, this works:
```
@profile
def fromApp(self, message, session_ID):
pass
```
But bug happens when I try to use it with the class:
```
@TimeWorker().profile
def fromApp(self, message, session_ID):
pass
```
**But currently I have the same problem even if I don't use it.**

**Comment by georgeivanyan** [comment]
I'm trying to narrow it down and find the source of the problem. I've got rid of the mentioned decorator and still see the problem. It's quite peculiar, as this step has previously helped me to resolve this. I have a specific py file in the project where the error always occurs whenever I simply open the file.
I've checked for additional details in the log and this is the error message that shows up: "Typeshed path not found ". I'm not sure if it's related but worth mentioning:

 ```
- 10:41:13 AM] (1140) Could not import '_functools' in file 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39\Lib\functools.py'
[Info  - 10:41:13 AM] (1140)   Looking in stubPath 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\typings'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\typings'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\typings'
[Info  - 10:41:13 AM] (1140)   Looking in root directory of execution environment 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39\DLLs'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39\DLLs'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39\DLLs'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39\Lib'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39\Lib'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39\Lib'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\AppData\Local\Programs\Python\Python39'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\win32'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\win32'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\win32'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\win32\lib'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\win32\lib'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\win32\lib'
[Info  - 10:41:13 AM] (1140)   Looking in python search path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\pythonwin'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\pythonwin'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\pythonwin'
[Info  - 10:41:13 AM] (1140)   Looking in bundled stubs path 'c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\bundled\stubs'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve stub package using root path 'c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\bundled\stubs'
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\bundled\stubs'
[Info  - 10:41:13 AM] (1140)   Looking for typeshed stdlib path
[Info  - 10:41:13 AM] (1140)   Looking for typeshed stdlib path
[Info  - 10:41:13 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\typeshed-fallback\stdlib'
[Info  - 10:41:13 AM] (1140)   Typeshed path not found
[Info  - 10:41:13 AM] (1140)   Looking for typeshed third-party path
[Info  - 10:41:13 AM] (1140)   Looking for typeshed stubs path
[Info  - 10:41:13 AM] (1140)   Typeshed path not found
[Info  - 10:52:04 AM] (1140) Could not import '.packages.six.moves.http_client' in file 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\urllib3\exceptions.py'
[Info  - 10:52:04 AM] (1140)   Attempting to resolve relative import
[Info  - 10:52:04 AM] (1140)   Attempting to resolve using root path 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\urllib3'
[Info  - 10:52:04 AM] (1140)   Resolved import with file 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\urllib3\packages\__init__.py'
[Info  - 10:52:04 AM] (1140)   Resolved import with file 'c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\.venv\Lib\site-packages\urllib3\packages\six.py'
```

Are there any specific steps you can suggest that I can follow to help narrow it down?

**Comment by erictraut** [comment]
Thanks for the additional logs. I don't see anything that's suspicious.

The technique I typically use to narrow down reproducible problems like this is to temporarily comment out code until the problem goes away. If it's a large file, I binary search it (comment out half the code, etc.) to speed up the search. This usually narrows it down pretty quickly to a single function or block.

**Comment by georgeivanyan** [comment]
I narrow it down to this part:

When I add or remove message_start variable, I get the error. 

```
import quickfix as fix
 
class FIX_application(fix.Application):
    
     def __init__(self, options):
        self.options = options
        message_start = f"Implementation was run with options:\nDB: {self.options.db}"

        super().__init__()
```

logfile:
```

[Error - 1:00:51 PM] (1140) An internal error occurred while type checking file "c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\src\data_application\fix_application copy.py": RangeError: Maximum call stack size exceeded
    at getDeclaredTypeForExpression (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:1995:14)
    at evaluateTypesForAssignmentStatement (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:13693:38)
    at evaluateTypesForStatement (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:16607:25)
    at callback (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17915:17)
    at evaluateTypeForSubnode (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:16708:13)
    at Object.getInferredTypeOfDeclaration (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17914:32)
    at getDeclaredCallBaseType (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1528:30)
    at isCallNoReturn (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1256:26)
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:995:25)
    at e (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1019:29)
(1140) [BG(1)]   checking: c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\src\data_application\fix_application copy.py (37ms)
(1140) [BG(1)] analyzing: c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\src\data_application\fix_application copy.py (38ms)
[Error - 1:00:51 PM] (1140) Error performing analysis: RangeError: Maximum call stack size exceeded
    at getDeclaredTypeForExpression (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:1995:14)
    at evaluateTypesForAssignmentStatement (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:13693:38)
    at evaluateTypesForStatement (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:16607:25)
    at callback (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17915:17)
    at evaluateTypeForSubnode (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:16708:13)
    at Object.getInferredTypeOfDeclaration (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\typeEvaluator.ts:17914:32)
    at getDeclaredCallBaseType (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1528:30)
    at isCallNoReturn (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1256:26)
    at isFlowNodeReachableRecursive (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:995:25)
    at e (c:\Users\GeorgeIvanyan\.vscode\extensions\ms-python.vscode-pylance-2022.2.1\dist\pyright-internal\src\analyzer\codeFlowEngine.ts:1019:29)
(1140) [BG(1)] indexing: c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\src\data_application\fix_application copy.py [found 1] (0ms)
(1140) Indexing Done: c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\src\data_application\fix_application copy.py
(1140) Background analysis message: getSemanticTokens delta
(1140) [BG(1)] getSemanticTokens delta previousResultId:1644602445077 at c:\Users\GeorgeIvanyan\OneDrive - EPFC\Projects\gi_projects\MKTXIntegration\src\data_application\fix_application copy.py (27ms)
[Error - 1:00:51 PM] Request textDocument/semanticTokens/full/delta failed.
  Message: Request textDocument/semanticTokens/full/delta failed unexpectedly without providing any details.
  Code: -32603 
```

Are there any other steps that can help you?

**Comment by erictraut** [comment]
Thanks! I'm able to repro the problem locally. I'll investigate further.

**Comment by DeusAres** [comment]
```
[Info  - 21:16:08] (7980) [BG(1)] Long operation: getSemanticTokens full at [redacted.py] (6515ms)
[Error - 21:10:25] Request textDocument/hover failed.
Message: Request textDocument/hover failed with message: Maximum call stack size exceeded
Code: -32603
 ```

I'm getting more or less the same error too, specifically, importing PySimpleGUI that is a pretty big library

```
import PySimpleGUI as sg

layout = [
    [sg.Text("Hello!")],
    [sg.Input("NICE!")],
    [sg.ColorChooserButton()] # This line is sufficient to create the error
]
```

@erictraut tag me if you need any kinda of log or report (w/ instruction to generate them), or if I should file a new issue

**Comment by erictraut** [comment]
Thanks @georgeivanyan and @DeusAres for the repros. Very helpful. I was able to find the root cause of the problem. It will be fixed in the next release.

**Comment by debonte** [comment]
This issue has been fixed in version 2022.2.3, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202223-16-February-2022)
